x-pack/legacy/plugins/lens/public/indexpattern_plugin/dimension_panel/format_selector.tsx

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiFormRow, EuiFieldNumber, EuiComboBox } from '@elastic/eui';
+import { KBN_FIELD_TYPES, ES_FIELD_TYPES } from 'src/plugins/data/common';
+import { IndexPattern } from '../types';
+import { IndexPatternColumn } from '../indexpattern';
+import { IndexPatternDimensionPanelProps } from './dimension_panel';
+
+const supportedFormats: Record<string, { title: string }> = {
+  number: {
+    title: i18n.translate('xpack.lens.indexPattern.numberFormatLabel', {
+      defaultMessage: 'Custom number',
+    }),
+  },
+  percent: {
+    title: i18n.translate('xpack.lens.indexPattern.percentFormatLabel', {
+      defaultMessage: 'Percent',
+    }),
+  },
+  bytes: {
+    title: i18n.translate('xpack.lens.indexPattern.bytesFormatLabel', { defaultMessage: 'Bytes' }),
+  },
+};
+
+type FormatSelectorProps = Pick<IndexPatternDimensionPanelProps, 'data'> & {
+  selectedColumn: IndexPatternColumn;
+  currentIndexPattern: IndexPattern;
+  onChange: (newFormat?: { id: string; params?: Record<string, unknown> }) => void;
+};
+
+export function FormatSelector(props: FormatSelectorProps) {
+  const { selectedColumn, data, onChange, currentIndexPattern } = props;
+
+  const fieldFormats = data.fieldFormats;
+  const selectedFormat = selectedColumn.format?.id
+    ? supportedFormats[selectedColumn.format.id]
+    : undefined;
+
+  let defaultFormatter;
+  const formatFromIndexpattern = currentIndexPattern.fieldFormatMap?.[selectedColumn.sourceField];
+  if (formatFromIndexpattern) {
+    defaultFormatter = fieldFormats.getType(formatFromIndexpattern.id);
+  } else {
+    // Look up the default formatter, such as number formatter
+    defaultFormatter = fieldFormats.getDefaultType(
+      selectedColumn.dataType as KBN_FIELD_TYPES,
+      (currentIndexPattern.fields.find(f => f.name === selectedColumn.sourceField)
+        ?.esTypes as ES_FIELD_TYPES[]) || []
+    );
+  }
+
+  const defaultOption = {
+    value: '',
+    label: i18n.translate('xpack.lens.indexPattern.defaultFormatLabel', {
+      defaultMessage: 'Default format ({defaultTitle})',
```

Comment:
Do we actually need to specify the default type in parentheses? It seems that it could be more confusing because of the case that I presented with the `$10` that is `Default (Number)`.

```suggestion
      defaultMessage: 'Default',
```